src/Http/Http.Extensions/gen/StaticRouteHandlerModel/StaticRouteHandlerModel.Parser.cs

```diff
@@ -0,0 +1,137 @@
+{
+    private const int _routePatternArgumentOrdinal = 1;
+    private const int _routeHandlerArgumentOrdinal = 2;
+
+    public static EndpointRoute GetEndpointRouteFromArgument(SyntaxToken routePattern)
+    {
+        return new EndpointRoute
+        {
+            RoutePattern = routePattern.ValueText
+        };
+    }
+
+    public static EndpointResponse GetEndpointResponseFromMethod(IMethodSymbol method)
+    {
+        return new EndpointResponse
+        {
+            ContentType = "plain/text",
+            ResponseType = method.ReturnType.ToString(),
+        };
+    }
+
+    public static Endpoint GetEndpointFromOperation(IInvocationOperation operation)
+    {
+        if (!TryGetRouteHandlerPattern(operation, out var routeToken))
+        {
+            return null;
+        }
+        if (!TryGetRouteHandlerMethod(operation, out var method))
+        {
+            return null;
+        }
+        var filePath = operation.Syntax.SyntaxTree.FilePath;
+        var span = operation.Syntax.SyntaxTree.GetLineSpan(operation.Syntax.Span);
+
+        var invocationExpression = (InvocationExpressionSyntax)operation.Syntax;
+        var httpMethod = ((IdentifierNameSyntax)((MemberAccessExpressionSyntax)invocationExpression.Expression).Name).Identifier.ValueText;
+
+        return new Endpoint
+        {
+            Route = GetEndpointRouteFromArgument(routeToken),
+            Response = GetEndpointResponseFromMethod(method),
+            Location = (filePath, span.EndLinePosition.Line + 1),
+            HttpMethod = httpMethod,
+        };
+    }
+
+    private static bool TryGetRouteHandlerPattern(IInvocationOperation invocation, out SyntaxToken token)
+    {
+        IArgumentOperation? argumentOperation = null;
+        foreach (var argument in invocation.Arguments)
+        {
+            if (argument.Parameter?.Ordinal == _routePatternArgumentOrdinal)
+            {
+                argumentOperation = argument;
+            }
+        }
+        if (argumentOperation?.Syntax is not ArgumentSyntax routePatternArgumentSyntax ||
+            routePatternArgumentSyntax.Expression is not LiteralExpressionSyntax routePatternArgumentLiteralSyntax)
+        {
+            token = default;
+            return false;
+        }
+        token = routePatternArgumentLiteralSyntax.Token;
+        return true;
+    }
+
+    private static bool TryGetRouteHandlerMethod(IInvocationOperation invocation, out IMethodSymbol method)
+    {
+        IArgumentOperation? argumentOperation = null;
+        method = null;
+        foreach (var argument in invocation.Arguments)
+        {
+            if (argument.Parameter?.Ordinal == _routeHandlerArgumentOrdinal)
+            {
+                argumentOperation = argument;
+            }
+        }
+
+        if (argumentOperation is not null)
+        {
+            method = ResolveMethodFromOperation(argumentOperation);
+            return true;
+        }
+
+        return false;
```

Comment:
A little bit of refactoring to simplify the code. Does this work and preserve the intended behavior?

```suggestion
        foreach (var argument in invocation.Arguments)
        {
            if (argument.Parameter?.Ordinal == _routeHandlerArgumentOrdinal)
            {
                method = ResolveMethodFromOperation(argument);
                return true;
            }
        }

        method = null;
        return false;
```